src/components/renderer/PageFormElements.tsx

```diff
@@ -39,45 +39,63 @@ function PageFormElements({
   const handleLookup = React.useCallback<FormElementLookupHandler>(
     (mergeLookupResults) => {
       setFormSubmission((currentFormSubmission) => {
-        const { submission, elements, executedLookups } = mergeLookupResults({
-          elements: pageElement.elements,
-          submission: currentFormSubmission.submission,
-          lastElementUpdated: currentFormSubmission.lastElementUpdated,
-          executedLookups: currentFormSubmission.executedLookups,
-        })
-
-        const definition = {
-          ...currentFormSubmission.definition,
-        }
         if (pageElement.id === formId.toString()) {
-          definition.elements = elements
-        } else {
-          definition.elements = currentFormSubmission.definition.elements.map(
-            (formElement) => {
-              if (
-                formElement.id === pageElement.id &&
-                formElement.type === 'page'
-              ) {
-                return {
-                  ...formElement,
-                  elements,
-                }
-              } else {
-                return formElement
-              }
+          const { submission, elements, executedLookups } = mergeLookupResults({
+            elements: currentFormSubmission.definition.elements,
+            submission: currentFormSubmission.submission,
+            lastElementUpdated: currentFormSubmission.lastElementUpdated,
+            executedLookups: currentFormSubmission.executedLookups,
+          })
+
+          return {
+            submission,
+            definition: {
+              ...currentFormSubmission.definition,
+              elements: elements,
```

Comment:
`elements` being passed from above will still have the dynamic stuff in it?

Reply:
These will be the elements stored in state, not the dynamic ones in the component tree.

Reply:
Ah its whatever your passing here that gets passed back.

Reply:
Alrighty then.